magicctapipe/scripts/lst1_magic/lst1_magic_cloud_profile_estimator.py

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env python
+"""
+This script uses geometrical model (see https://doi.org/10.1016/j.jheap.2024.03.003
+for details) for the estimation of the vertical transmission, base height and thickness
+of a cloud, the so-called "proton" LIDAR.
+
+The script calculates an offset angle from the primary direction, corresponding to the
+emission height of Cherenkov light. This is necessary to construct the longitudinal
+profile of the observed Cherenkov light and estimate the cloud's vertical transmission
+profile. It processes DL1 stereo files, generating output DL1 stereo files that also
+contain the longitudinal  distribution of the observed Cherenkov light.
+
+Note: DL1 stereo files must include images. This requires processing the DL1 files to the
+stereo level with the "save_images: True" flag enabled in the configuration file, with
+the following quality cuts:
+
+quality_cuts: "(n_pixels > 20) & (n_islands < 1.5) & (concentration_cog > 0.001) & ((slope < -1)| (slope > 1))"
+
+Usage:
+$ python lst1_magic_cloud_profile_estimator.py
+--input_file "/user/data/dl1_stereo_LST-1_MAGIC.Run03265.0040.h5"
+(--output_dir "/user/data/cloud_profiles/")
+(--config_file "/user/data/config.yaml")
+"""
+
+import argparse
+import logging
+import time
+from pathlib import Path
+
+import astropy.units as u
+import ctapipe
+import h5py
+import numpy as np
+import pandas as pd
+import tables
+import yaml
+from astropy.coordinates import AltAz, SkyCoord
+from ctapipe.coordinates import TelescopeFrame
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.io import read_table
+
+import magicctapipe
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def model(imp, h, alt):
+    """
+    Geometrical model needed to estimate the vertical tranmission profile of a cloud.
+    Calculates an oﬀset angle from the primary direction corresponding
+    to the emission height of the Cherenkov light need to construct the longitudinal
+    profile of the observed Cherenkov light
+    and to estimate the vertical tranmission profile of a cloud.
+
+    Parameters
+    ----------
+    imp : float
```

Comment:
imp and h are actually not floats but Quantity objects (you can see in line 75 that we convert the ratio to a number without a unit